test/libsolidity/util/BytesUtils.cpp

```diff
@@ -164,35 +201,105 @@ string BytesUtils::formatString(bytes const& _bytes) const
 	return os.str();
 }
 
-bytes BytesUtils::alignLeft(bytes _bytes) const
+string BytesUtils::formatRawBytes(bytes const& _bytes)
 {
-	return std::move(_bytes) + bytes(32 - _bytes.size(), 0);
+	if (!_bytes.empty())
+	{
+		stringstream os;
+		auto it = _bytes.begin();
+		size_t size = 32;
+		for (size_t i = 0; i < _bytes.size() / size; i++)
+		{
+			long offset = static_cast<long>(size);
+			auto offsetIter = it + offset;
+			bytes byteRange{it, offsetIter};
+
+			os << "  " << byteRange;
+
+			it += offset;
+			if (it != _bytes.end())
+				os << endl;
+		}
+
+		return os.str();
+	}
+	else
+		return "[]";
 }
 
-bytes BytesUtils::alignRight(bytes _bytes) const
+string BytesUtils::formatBytes(
+	bytes const& _bytes,
+	ABIType const& _abiType
+)
 {
-	return bytes(32 - _bytes.size(), 0) + std::move(_bytes);
+	stringstream os;
+
+	switch (_abiType.type)
+	{
+	case ABIType::UnsignedDec:
+		// Check if the detected type was wrong and if this could
+		// be signed. If an unsigned was detected in the expectations,
+		// but the actual result returned a signed, it would be formatted
+		// incorrectly.
+		if (*_bytes.begin() & 0x80)
+			os << formatSigned(_bytes);
+		else
+			os << formatUnsigned(_bytes);
+		break;
+	case ABIType::SignedDec:
+		os << formatSigned(_bytes);
+		break;
+	case ABIType::Boolean:
+		os << formatBoolean(_bytes);
+		break;
+	case ABIType::Hex:
+		os << formatHex(_bytes);
+		break;
+	case ABIType::HexString:
+		os << formatHexString(_bytes);
+		break;
+	case ABIType::String:
+		os << formatString(_bytes);
+		break;
+	case ABIType::Failure:
+		break;
+	case ABIType::None:
+		break;
+	}
+	return os.str();
 }
 
-bytes BytesUtils::applyAlign(
-	Parameter::Alignment _alignment,
-	ABIType& _abiType,
-	bytes _bytes
-) const
+string BytesUtils::formatBytesRange(
+	bytes _bytes,
+	dev::solidity::test::ParameterList const& _parameters,
+	bool _highlight
+)
 {
-	if (_alignment != Parameter::Alignment::None)
-		_abiType.alignDeclared = true;
+	stringstream os;
+	auto it = _bytes.begin();
 
-	switch (_alignment)
+	for (auto const& parameter: _parameters)
 	{
-	case Parameter::Alignment::Left:
-		_abiType.align = ABIType::AlignLeft;
-		return alignLeft(std::move(_bytes));
-	case Parameter::Alignment::Right:
-		_abiType.align = ABIType::AlignRight;
-		return alignRight(std::move(_bytes));
-	default:
-		_abiType.align = ABIType::AlignRight;
-		return alignRight(std::move(_bytes));
+		size_t size = parameter.abiType.size;
```

Comment:
I think this could be better readable if we would not have the local variables `size` and `offset`. Also please try to avoid the cast.

Reply:
I removed the local variable but left the cast to `long`, which is needed in order to avoid a warning about the signedness of the result of `it + type.size`, which is a `size_t` being implicitly changed. I don't think that the implicit conversion is dangerous and would be also OK with introducing a new warning.